### PR TITLE
docs(augmentation): reconcile the batch 6d Convention blocks and pins with the #4649, #4655 and #4656 fixes

### PR DESCRIPTION
**Pull request by ducha-aiki**
`main` is red since #4634 (batch 6d) merged: three repair PRs landed between its base and its merge — #4677 (RandAugment posterize / translate units), #4678 (the unused `OperationBase` probability sampler) and #4685 (the double `p` gate in `RandomMixUpV2` / `RandomCutMixV2`) — and 6d's Convention blocks and wart pins still describe the behavior those PRs repaired. Four pins fail on `main` @ `c90c05f66`:

```
tests/augmentation/test_conventions_auto.py::TestAutoAugmentConventions::test_wart_randaugment_posterize_and_translate_units_4655
tests/augmentation/test_conventions_auto.py::TestAutoAugmentConventions::test_wart_operation_probability_parameter_is_inert_4656
tests/augmentation/test_conventions_auto.py::TestAutoAugmentConventions::test_convention_randaugment_formula_holds_for_every_default_entry
tests/augmentation/test_conventions_mix.py::TestMixConventions::test_wart_mixup_and_cutmix_apply_p_to_rows_a_second_time_4649
```

This change is docstrings, one `.rst` paragraph and tests; no runtime code moves.

**Docstrings** — the sentences that cited issues 4649, 4655 and 4656 as open defects are rewritten to the repaired behavior, each claim executed on this branch:
- `RandomMixUpV2` / `RandomCutMixV2`: `p` is applied once, as one batch-wide draw; `_params["batch_prob"]` is all ones or all zeros (300 seeds, only those two patterns occur) and no row of a selected batch is dropped again.
- `RandAugment`: `posterize` runs its range backwards and truncates (default `("posterize", 4.0, 8.0)` keeps 7 / 6 / 4 bits at `m` = 7 / 15 / 29); `translate_x` / `translate_y` ranges are fractions of the width / height (`m=15` on `(-0.1, 0.1)` moves a 32-pixel side by 1.6 px). The `shear_x` / `shear_y` ×180 mapping is unchanged and stays documented (27°).
- `OperationBase`: the probability parameter stays in `state_dict()` for checkpoint compatibility, takes no part in sampling and receives no gradient (the magnitude does); no sampler is stored, so `copy.deepcopy` works on all three policies fresh, after a forward, and after `train()` / `eval()`, and the copy replays the original's `_params` bitwise. `PolicySequential` keeps its pointer to the still-open magnitude-bypass issue #4441. The Conventions page's serialization paragraph says the same.

**Pins** — the three wart pins are inverted into `test_convention_*` pins on the repaired behavior (each is red with the corresponding repair reverted, since it asserts the opposite of the wart it replaces), and the formula pin gains the backwards posterize range and the per-side translate scaling, on an `H != W` shape so x and y cannot be confused.

**CutMix `same_on_batch`** — the Convention block states what is measured: with `same_on_batch=False` each row and mix draws its own cut size, but the placement is one uniform draw per axis shared by all of them (filed as #4712, a regression from the #1459 generator refactor); `same_on_batch=True` shares the size as well. `test_convention_cutmix_same_on_batch_shares_cut_geometry` uses the default `cut_size` so that sharing is what it measures (it is red with the generator ignoring the flag), and `test_wart_cutmix_placement_is_one_shared_draw_4712` pins the open defect.

Checked: the five batch-6 convention files under `--dtype=float32,float64,float16,bfloat16` — 403 passed, 4 xfailed (the pre-existing strict xfails); `pre-commit` on the changed files; `grep -rnE "#(4649|4655|4656)|issues/(4649|4655|4656)" kornia/ docs/source` is empty, and the remaining `tests/` hits are the inverted pins and #4685's own regression tests. torch 2.14.0, cpu, macOS arm64; CUDA, MPS and the 2.5.1 floor are left to CI.

No changelog fragment: documentation and test reconciliation only (`no-changelog`). Part of the batch 6 close-out, tracking issue #4407.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

### CODE REVIEW

**Comment by ducha-aiki** [comment]
Intended verdict: **REQUEST_CHANGES**

[P2] `kornia/augmentation/_2d/mix/cutmix.py:81-84` says that every row and mix of a selected batch draws its own cut. That is not true for the public `same_on_batch=True` mode: `CutmixGenerator.forward` shares the beta-derived size and start position across the flattened `B * num_mix` samples. On this head, `num_mix=2` and `B=8` produce one unique `crop_src` among all 16 cuts. Please qualify the convention (independent cuts only when `same_on_batch=False`; shared geometry across rows and mixes when `True`) and preferably pin that mode in `test_conventions_mix.py`.

The remaining documentation updates match the merged fixes. Local validation passed: 94 focused convention tests and the full pre-commit suite. Read the Docs and pre-commit CI are green; broader CI is still pending.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5).

**Comment by ducha-aiki** [comment]
## Review at `cc6f146f9`, then pushed `9be2f1168`

**Verdict: approve at `9be2f1168`, once CI is green.** Two P2 items found in `cc6f146f9`; both are fixed in the pushed commit, so nothing is left open here.

### What I verified (torch 2.14.0, CPU, macOS arm64, from a worktree at the PR head; `kornia.__file__` checked)

- The four pins named in the description are red on `main` @ `c90c05f66` and green here. `test_conventions_auto.py`, `test_conventions_mix.py`, `test_conventions_3d.py` under `--dtype=float32,float64,float16,bfloat16`: 290 passed, 4 xfailed.
- **Every inverted pin can fail.** Reverting the library part of each repair on this head:
  | mutation | result |
  |---|---|
  | revert #4685 (`a04938fac`) | 1 failed |
  | revert #4678 (`237011b72`) | 1 failed |
  | revert #4677 (`9d45bd145`) | 2 failed (units pin + formula pin) |
  | only MixUp's generator back to `p=p` | 1 failed |
  | only CutMix's generator back to `p=p` | 1 failed |
  | `translate_x` scaled by the height instead of the width | 2 failed — the `H != W` shapes do their job |
  | posterize range forwards again | 2 failed |
- Docstring claims read against the repaired code: `p=1.0` is what both mix classes now hand their generators; the posterize `7 / 6 / 4` bits and the `1.6` px translate are the values the pins assert; `_probability` is initialised from `p` / `p_batch`, stays in `state_dict()`, gets no gradient while `_magnitude` does; `deepcopy` replay is bitwise on all three policies.
- Closing grep `#(4649|4655|4656)|issues/(4649|4655|4656)` over `kornia/` and `docs/source`: empty. The two `tests/` hits are #4685's own regression tests. A phrase grep for the un-numbered leftovers (`p ** 2`, `RelaxedBernoulli`, `graph leaves`, `all-black`, `as pixels`, `probability sampler`) finds nothing stale.
- #4649, #4655, #4656 are closed; #4441 and #4469, which the text still points to, are open. No closing keyword in the commit messages or the body.

### [P2, fixed] The `same_on_batch` sharing pin could not fail

`test_convention_cutmix_same_on_batch_shares_cut_geometry` used `cut_size=(0.5, 0.5)`. Running its body with `same_on_batch=False` gives an all-equal `crop_src` in **200 of 200 seeds**, so the assertion held with or without the flag. The reason is a library defect, not the fixture alone: `CutmixGenerator` draws the cut placement as **one** uniform number per axis, shared by every row and mix, in both modes; only the sizes are per-row. On a 100000-px image with `same_on_batch=False` and eight cuts, `x0 / (W - w - 1)` is `0.0340` for all eight while the widths range over 51559–95768. The original #609 code drew the placement per row; the #1459 generator refactor collapsed it. Filed as #4712.

That also made the new docstring sentence ("with `same_on_batch=False` the cut geometries are not required to match") too vague to be checkable. In `9be2f1168`:
- the Convention block says what is measured: per-row sizes, one shared placement draw per axis (#4712), and `same_on_batch=True` sharing the size as well;
- the sharing pin uses the default `cut_size` and additionally asserts that `same_on_batch=False` yields differing heights and widths. With the generator forced to ignore `same_on_batch`, it fails;
- `test_wart_cutmix_placement_is_one_shared_draw_4712` pins the defect on a fixed size. With a per-row placement draw planted in the generator it fails and the convention pin stays green, so the #4712 fix has exactly one pin to invert.

### [P2, fixed] "stores no sampler … on the wrapped augmentation" (Copilot's open thread)

Valid as read: the wrapped augmentation keeps its parameter samplers in `_param_generator`, and `deepcopy` works with them. Only the wrapper's probability sampler went away in #4678. Reworded to "builds no probability sampler, on itself or on the wrapped augmentation, which keeps its own parameter samplers". The pin already asserts the narrow claim (`_p_gen` / `_p_batch_gen` absent), so no test change.

### After the push

Convention and mix suites under four dtypes: 811 passed. Doctests of the two touched modules pass. `pre-commit` on the changed files passes. `merge-tree` against `main`, #4708 and #4506 (the open PRs sharing a file): no conflict in any file of this PR. Not verified locally: CUDA, MPS, the torch 2.5.1 floor and the `-W` docs build — left to CI.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

File: kornia/augmentation/_2d/mix/cutmix.py [hunk 1]
```diff
@@ -78,11 +78,10 @@ class RandomCutMixV2(MixAugmentationBaseV2):
...
           the compatibility default returns ``cut_area / image_area`` and emits a deprecation warning.
-        - ``p`` is a batch-wide gate, and the generator applies it a second time per row and mix: inside a selected
-          batch each cut is kept with probability ``p`` and zeroed otherwise. Each cut passes both gates with
-          probability ``p ** 2``; it can still leave the image unchanged through self-pairing or a zero-sized cut.
-          With ``same_on_batch=True`` the inner gate is shared across rows and mixes
-          (`#4649 <https://github.com/kornia/kornia/issues/4649>`_).
+        - ``p`` is a batch-wide gate and is applied once: one draw per call selects the whole batch with probability
+          ``p``, so ``_params["batch_prob"]`` is all ones or all zeros. Every row and mix of a selected batch draws
+          its own cut and none is dropped again; a cut can still leave the image unchanged through self-pairing or a
+          zero-sized cut.
```
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/augmentation/_2d/mix/cutmix.py]: Addressed in `9be2f1168` — details in the review comment on this PR.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

File: kornia/augmentation/auto/operations/base.py [hunk 2]
```diff
@@ -41,11 +41,14 @@ class OperationBase(nn.Module):
...
-          receives no gradient (`#4656 <https://github.com/kornia/kornia/issues/4656>`_). It then draws the wrapped
-          augmentation's parameters and substitutes the supplied or learned magnitude.
+          the wrapped generator's configured range. The probability parameter is initialized from the wrapped
+          augmentation's ``p`` (``p_batch`` for a batch operation) and kept in ``state_dict()`` for API and
+          checkpoint compatibility; it does not take part in sampling. The gate is drawn from the wrapped
+          augmentation's own float ``p`` and ``p_batch``, the sampled ``batch_prob`` is a hard ``0`` or ``1``, and
+          the probability parameter receives no gradient, while the magnitude does. ``forward_parameters`` draws
+          the wrapped augmentation's parameters and substitutes the supplied or learned magnitude. The wrapper
+          stores no sampler on itself or on the wrapped augmentation, so ``copy.deepcopy`` works before and after a
+          forward pass or a ``train()`` / ``eval()`` call.
```
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/augmentation/auto/operations/base.py]: Addressed in `9be2f1168` — details in the review comment on this PR.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
